lib/rules/space-infix-ops.js

```diff
@@ -54,15 +46,17 @@ module.exports = {
         function getFirstNonSpacedToken(left, right) {
             const tokens = sourceCode.getTokensBetween(left, right, 1);
 
-            for (let i = 1, l = tokens.length - 1; i < l; ++i) {
-                const op = tokens[i];
+            for (let i = 1; i < tokens.length - 1; i++) {
+                const prev = tokens[i - 1];
+                const curr = tokens[i];
+                const next = tokens[i + 1];
+
+                if (curr.value === "(" || curr.value === ")") {
```

Comment:
You mentioned that:

> there can be a lot of tokens between `left` and `right`, but only one of them can be an infix operator, the others must be opening and closing parens.

Off the top of my head, I can't think of any cases where this would be wrong, although it's possible that there are some cases that we haven't thought of.

To make sure it's correct, would it be possible to add the expected operator as a parameter to `getFirstNonSpacedToken`? For example, `checkVar` could use something like `getFirstNonSpacedToken(leftNode, rightNode, "=")`. Then the implementation of `getFirstNonSpacedToken` could look something like:

```js
function getFirstNonSpacedToken(left, right, operatorValue) {
    const operator = sourceCode.getTokenBetween(
        left,
        right,
        token => token.value === operatorValue
    );
    const beforeOperator = sourceCode.getTokenBefore(operator);
    const afterOperator = sourceCode.getTokenAfter(operator);
    // ...check to see if there's a space, as before
}
```

Reply:
nice idea, i think it might fix #10922 as well ✊